### PR TITLE
More nullable reference type updates

### PR DESCRIPTION
**Pull request by slewis74**
As part of working on getting the newer versions (NetCore) of Calamari to run on newer versions of Windows a few more nullable reference type false positives came up. This PR is to address them.

### CODE REVIEW

File: source/Calamari.Common/Plumbing/Variables/IVariables.cs [hunk 1]
```diff
@@ -10,13 +10,17 @@ public interface IVariables : IEnumerable<KeyValuePair<string, string>>
...
+        string? GetRaw(string variableName);
         [return: NotNullIfNotNull("defaultValue")]
         string? Get(string variableName, string? defaultValue = null);
         [return: NotNullIfNotNull("defaultValue")]
-        string? Get(string variableName, out string error, string? defaultValue = null);
-        string Evaluate(string expressionOrVariableOrText, out string error, bool haltOnError = true);
-        string Evaluate(string expressionOrVariableOrText);
+        string? Get(string variableName, out string? error, string? defaultValue = null);
+
+        [return: NotNullIfNotNull("expressionOrVariableOrText")]
```
Comment (benPearce1) [inline source/Calamari.Common/Plumbing/Variables/IVariables.cs:19]: This sucks a bit, looks like .Net Core will improve this in the future https://github.com/dotnet/csharplang/issues/373

File: source/Calamari.Common/Features/StructuredVariables/XmlFormatVariableReplacer.cs [hunk 2]
```diff
@@ -61,6 +61,8 @@ public void ModifyFile(string filePath, IVariables variables)
                 {
                     var selectedNodes = navigator.XPath2SelectNodes(xPathExpression, nsManager);
                     var variableValue = variables.Get(variable.Key);
+                    if (variableValue == null)
+                        continue;
```
Comment (benPearce1) [inline source/Calamari.Common/Features/StructuredVariables/XmlFormatVariableReplacer.cs:65]: How do we want to handle empty strings here?

**Review by benPearce1 — APPROVED** [review]
👍 
no functional changes I can see. One question.

File: source/Calamari.Common/Plumbing/Variables/IVariables.cs (hunk 1 above)
Comment (slewis74) [reply source/Calamari.Common/Plumbing/Variables/IVariables.cs:19]: yes, I did try `nameof`, just to be sure it didn't work :)

File: source/Calamari.Common/Features/StructuredVariables/XmlFormatVariableReplacer.cs (hunk 2 above)
Comment (slewis74) [reply source/Calamari.Common/Features/StructuredVariables/XmlFormatVariableReplacer.cs]: good question. I've revisited and put it back to assigning the nulls, the Xml objects all seems to deal with that correctly and then we don't risk accidentally changing behavior

**Pull request merged**